python/cudf/cudf/core/dataframe.py

```diff
@@ -7011,15 +7017,22 @@ def _apply_cupy_method_axis_1(self, method, *args, **kwargs):
         if skipna is not False and method in _cupy_nan_methods_map:
             method = _cupy_nan_methods_map[method]
 
+        if len(arr) == 0 and method == "nanmedian":
+            method = "median"
```

Comment:
Is this working around a cupy limitation?

Reply:
Yes